### ISSUE TITLE
using alternative HTTP clients

### ISSUE THREAD

**Issue by benswift** [cbh123/replicate-elixir#29]
**tl;dr** are you open to a PR which replaces the HTTPoison dep with Req? 

The "my app uses a different HTTP client to one of my deps, and leads to bloat" issue is obviously a tricky one. Andrea Leopardi has the best summary of [the current state of play for Elixir HTTP clients](https://andrealeopardi.com/posts/breakdown-of-http-clients-in-elixir/).

In particular, httpoison is a bit old-school these days - many cool kids are using [Req](https://hexdocs.pm/req/readme.html) (my fav) or [Tesla](https://hexdocs.pm/tesla/readme.html).

My specific question in this issue: is this lib using httpoison for a specific reason? (There used to be some multipart form-based reasons to use it over e.g. Req, although I don't think that's still the case.) The Leopardi blog post linked above says that other HTTP client libs have superseded httpoison these days. If not, are you open to a PR which replaces the HTTPoison dep with Req? Client API would remain the same; winners of this change would be those who are already using Req in their apps, and losers would be those who are currently using  httppoison (in my estimation the former is a bigger group than the latter, although I don't have stats on that). Although no-one's code would break, the "victims of bloat" problem would just be re-distributed.

Anyway, sorry about the long issue - just wanted to check first before I did any work. If you don't want it that's fine too :)

**Comment by benswift** [comment]
Even Jose [seems to be thinking along these lines for Phoenix](https://github.com/phoenixframework/phoenix/issues/5633#issue-1990103190).
